gadget/gadget.go

```diff
@@ -255,6 +259,16 @@ type VolumeStructure struct {
 	EnclosingVolume *Volume `yaml:"-" json:"-"`
 }
 
+// SetEnclosingVolumeInStructs is a helper that sets the pointer to
+// the Volume in all VolumeStructure objects it contains.
+func SetEnclosingVolumeInStructs(vv map[string]*Volume) {
```

Comment:
this can probably be called just SetEnclosingVolumes, if you think we need the InStructs, then I think it needs to be expanded to the full "InStructures",  as Struct makes one thing of struct types, like reflect.Struct